Source/WebCore/platform/graphics/filters/FilterImage.cpp

```diff
@@ -153,11 +155,17 @@ static void copyImageBytes(const PixelBuffer& sourcePixelBuffer, PixelBuffer& de
     if (destinationRect.isEmpty())
         return;
 
-    int size = sourceRectClipped.width() * 4;
-    int destinationBytesPerRow = destinationPixelBufferRect.width() * 4;
-    int sourceBytesPerRow = sourcePixelBufferRect.width() * 4;
-    uint8_t* destinationPixel = destinationPixelBuffer.bytes() + destinationRect.y() * destinationBytesPerRow + destinationRect.x() * 4;
-    const uint8_t* sourcePixel = sourcePixelBuffer.bytes() + sourceRectClipped.y() * sourceBytesPerRow + sourceRectClipped.x() * 4;
+    auto size = CheckedUint32(sourceRectClipped.width()) * 4;
+    auto destinationBytesPerRow = CheckedUint32(destinationPixelBufferRect.width()) * 4;
+    auto sourceBytesPerRow = CheckedUint32(sourcePixelBufferRect.width()) * 4;
+    auto destinationOffset = destinationRect.y() * destinationBytesPerRow + CheckedUint32(destinationRect.x()) * 4;
+    auto sourceOffset = sourceRectClipped.y() * sourceBytesPerRow + CheckedUint32(sourceRectClipped.x()) * 4;
+
+    if (size.hasOverflowed() || destinationBytesPerRow.hasOverflowed() || sourceBytesPerRow.hasOverflowed() || destinationOffset.hasOverflowed() || sourceOffset.hasOverflowed())
```

Comment:
I think we can use UNLIKELY here since the overflow case is very rare.

`if (UNLIKELY(size.hasOverflowed() || ...))`

Reply:
Fixed.